docs/script_convert.py

```diff
@@ -0,0 +1,75 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import argparse
+import pathlib
+
+BASH = "# bash"
+BASH_IGNORE = "# bash-ignore"
+BASH_MULTILINE_COMMENT_START = ": '"
+BASH_MULTILINE_COMMENT_END = "'"
+
+
+def convert(args: argparse.Namespace):
+    src_path = pathlib.Path(args.script)
+    des_path = src_path.parent / f"{src_path.stem}.py"
+    with open(src_path, "r") as src_f:
+        with open(des_path, "w") as des_f:
+            line = src_f.readline()
```

Comment:
i'd suggest you extract the contents of this `with` to a separate function so it can be tested

Reply:
can you clarify more. thanks!

Reply:
what i'm saying is if you create a function `def convert_script(src_f, des_f):`, then you can unit test it with BytesIO in place of real files

Reply:
done.